packages/components/Section.tsx

```diff
@@ -3,6 +3,7 @@ import { ReactNode } from 'react';
 
 interface Props extends ChakraProps {
   children: ReactNode;
+  id?: string;
```

Comment:
Porque usa essa ? após o id?

Reply:
também não entendi essa sintaxe.

Reply:
Bom, vamos dar um passo atrás:

1. Estou criando uma `interface` para o objeto `props`. Isso significa que estou criando o formato dele.
2. Algumas vezes queremos fazer com que uma propriedade seja opcional, ou seja, que seja tenha um valor especificado (no caso `string`) ou `undefined` se nada for passado.

No TypeScript temos duas formas de falar que uma chave de um elemento é opcional:

```ts
interface Props {
  // ...
  id: undefined | string
}

// ou

interface Props {
  // ...
  id?: string
}
```

Dessa forma o `id` pode ser passado ou não.

Links:
- https://www.typescriptlang.org/docs/handbook
- https://www.typescriptlang.org/docs/handbook/2/objects.html
- https://www.typescriptlang.org/docs/handbook/2/functions.html#optional-parameters